### PR TITLE
Move log_normalize from hdpmodel to matutils and add feature normalization capability

### PR DESCRIPTION
**Pull request by ArifAhmed1995**
- Log normalizing may be used for other models too, hence it's better to shift it to matutils
- log normalize : Currently for an input 2D array , independently normalizing each sample is done.An optional axis parameter is introduced to give support for normalizing along feature.

See axis parameter in : 
http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.normalize.html

### CODE REVIEW

**Comment by tmylk** [comment]
Please remove the unnecessary commits by running 'git fetch' and then 'git merge develop'

**Comment by ArifAhmed1995** [comment]
@tmylk #950
